app/workers/step_by_step_draft_update_worker.rb

```diff
@@ -23,17 +23,17 @@ def update_draft
   end
 
   def update_assigned_to
-    unless step_by_step_page.has_draft?
-      generate_internal_change_note
+    unless step_by_step_page.assigned_to == @current_user
```

Comment:
Nitpick: You could move this check to a function e.g. `assigned_to_current_user?`